errors/OverlappingPattern.md

```diff
@@ -0,0 +1,22 @@
+# `OverlappingPattern` Error
+
+## Example
+
+```purescript
+not :: Boolean -> Boolean
+not true  = false
+not _     = true
+not false = true
+```
+
+## Cause
+
+This error occurs when a pattern matching definition has **redundant** patterns, i.e., overlapping clauses.
+
+In the example, notice that the third case is redundant, as it is covered by the second case (which covers all remaining cases). That code is unreachable.
```

Comment:
Perhaps "the third case is unreachable" here? It might not be clear what "that" refers to in the last sentence.